### DISCUSSION TITLE
run bash script inside meson.build with run_command

### DISCUSSION THREAD

**Discussion by 5wattbulb** [mesonbuild/meson#9779] [Q&A] [answered]
r=run_command('ls') works fine however if I create an executable file called scpt containing 'ls' and use

`r=run_command(scpt)
`
inside meson.build I get a list of python (I think) messages with

`OSError: [Errno 8] Exec format error: '/home/tom/dev/pj/scpt'
`
tagged on at the end.

Running '/home/tom/dev/pj/scpt' in bash also works as expected.
Where am I going wrong?

**Comment by 5wattbulb** [comment, marked as answer]
Finally got this to work by inserting
`#!/bin/bash`
at the beginning of the script.

**Reply by andy5995** [reply]
You want to change `bash` to `sh` (probably no reason to specify bash when any ole interpreter will do).

Here's [a set-up for using scripts to test a binary app](https://github.com/theimpossibleastronaut/rmw/blob/b0ba6a423b22f497adab117ec8d576897919cac4/test/meson.build#L41) for anyone who might be interested in such a thing.

**Reply by 5wattbulb** [reply]
Thanks Andy. Useful.
